GCRCatalogs/catalog_configs/dc2_object_run2.2i_dr6_wfd_matched_addon.yaml

```diff
@@ -1,13 +1,20 @@
-#  Use ^/ to indicate file path relative to GCR root dir
-
-subclass_name: composite.CompositeReader
-description: DC2 Run 2.2i Object Catalog with Matched Table addon
-creators: ['Javi Sanchez', 'Eve Kovacs']
+based_on: dc2_object_run2.2i_dr6_v1_with_addons
 catalogs:
   - catalog_name: dc2_object_run2.2i_dr6_wfd_v1
```

Comment:
Maybe a separate issue from this PR, but it seems that the `deprecated` key is ignored for catalogs which are either composites or use `based_on`

While investigating I came across two catalogs whose `deprecated` messages reference catalogs which no longer exist:
`dc2_object_run2.1i_dr1_tract3830`
`dc2_object_run2.1i_dr4_tract3830`

Reply:
The `deprecated` key indeed only works if it appears in the top level. This is by design rather than a bug. The reason behind is to make its behavior similar to `include_in_default_catalog_list`. 

The two `_tract3830` catalogs should not exist in the first place too. Maybe instead of deleting it I should deprecate them instead.

Reply:
It seems to be the opposite  - it doesn't work at the top level, but I did see messages when referenced deprecated catalogs internally